pkg/exec/exec.go

```diff
@@ -18,6 +19,19 @@ import (
 	"k8s.io/klog/v2"
 )
 
+// WrapCommandInShell wraps the given command args in a shell invocation (sh -c "...")
+// This is useful when the command contains shell features like pipes, &&, ||, etc.
+// When using this function, users are expected to pass their command as a single quoted
+// string (e.g., "cd /app && pip install -r requirements.txt") which will be interpreted
+// by the shell. Multiple args are joined with spaces.
+func WrapCommandInShell(commandArgs []string) []string {
+	if len(commandArgs) == 0 {
+		return commandArgs
+	}
+	// Join all args with spaces and wrap with sh -c
+	return []string{"sh", "-c", strings.Join(commandArgs, " ")}
```

Comment:
@samos123 it defaults to `sh` , do you think people will require to override it with bash in the future?